db/src/types.rs

```diff
@@ -24,6 +24,7 @@ pub struct PeerData {
     pub last_seen: String,
     pub country: String,
     pub city: String,
+    pub synced: Option<bool>,
```

Comment:
if it's only a `bool` how should we set the field on the database? false by default?

Reply:
> if it's only a `bool` how should we set the field on the database? false by default?

Yes, or at least it was my understanding that you can't insert into the database until `PeerData` is constructed, and it is only ever constructed after we determine whether or not the peer is synced (when `synced` is set to `Some`). I guess `None` is valid if there is no working RPC though, since there would be no way to know if the peer is synced.

Reply:
exactly. So I used `Option<bool>` just in order to be able to represent the fact that we don't know if that node is synced or not since we weren't able to get this information, so we just save `None` which gets translated into a `NULL`